src/libraries/System.Globalization/tests/System/Globalization/CharUnicodeInfoTests.Generated.cs

```diff
@@ -78,6 +79,59 @@ public void GetUnicodeCategory_Int32()
             }
         }
 
+        [Fact]
+        public void TestCasing()
+        {
+            Func<uint, uint> toUpperUInt = (Func<uint, uint>) typeof(CharUnicodeInfo)
+                                                                .GetMethod("ToUpper", BindingFlags.NonPublic | System.Reflection.BindingFlags.Static, new Type[] { typeof(uint) })
+                                                                .CreateDelegate(typeof(Func<uint, uint>));
+            Func<char, char> toUpperChar = (Func<char, char>) typeof(CharUnicodeInfo)
+                                                                .GetMethod("ToUpper", BindingFlags.NonPublic | System.Reflection.BindingFlags.Static, new Type[] { typeof(char) })
+                                                                .CreateDelegate(typeof(Func<char, char>));
+            Func<uint, uint> toLowerUInt = (Func<uint, uint>) typeof(CharUnicodeInfo)
+                                                                .GetMethod("ToLower", BindingFlags.NonPublic | System.Reflection.BindingFlags.Static, new Type[] { typeof(uint) })
+                                                                .CreateDelegate(typeof(Func<uint, uint>));
+            Func<char, char> toLowerChar = (Func<char, char>) typeof(CharUnicodeInfo)
+                                                                .GetMethod("ToLower", BindingFlags.NonPublic | System.Reflection.BindingFlags.Static, new Type[] { typeof(char) })
+                                                                .CreateDelegate(typeof(Func<char, char>));
+
+            for (int i = 0; i <= 0xFFFF; i++)
+            {
+                if ((i >= 0xD800 && i <= 0xDFFF) || // Surrogates are characters in the Unicode range U+D800—U+DFFF
```

Comment:
See earlier comment. Instead of calling _continue_ in these cases, we should special-case what the return values actually should be, and test for those return values specifically. Otherwise these chars go untested.